Sources/Client/Client/Client.swift

```diff
@@ -199,7 +194,19 @@ public final class Client: Uploader {
         logger = logOptions.logger(icon: "🐴", for: [.requestsError, .requests, .requestsInfo])
         
         self.defaultURLSessionConfiguration = defaultURLSessionConfiguration
-        self.defaultWebSocketProviderType = defaultWebSocketProviderType
+        
+        if let defaultWebSocketProviderType = defaultWebSocketProviderType {
+            self.defaultWebSocketProviderType = defaultWebSocketProviderType
+        } else if #available(iOS 13, *) {
+            switch config.defaultWebSocketProviderType {
+            case .native:
+                self.defaultWebSocketProviderType = URLSessionWebSocketProvider.self
+            case .starscream:
+                self.defaultWebSocketProviderType = StarscreamWebSocketProvider.self
+            }
+        } else {
+            self.defaultWebSocketProviderType = StarscreamWebSocketProvider.self
+        }
```

Comment:
Do we need this logic here? I think it's OK to use just the value from settings. on iOS13< there will be just one value in the enum.

Reply:
Can you clarify?

Reply:
What I meant is that the `#available` check is not needed, because `webSocketProviderType` already have the iOS13 check.

Reply:
You mean this?
```swift
if let defaultWebSocketProviderType = defaultWebSocketProviderType {
    self.defaultWebSocketProviderType = defaultWebSocketProviderType
} else {
    switch config.defaultWebSocketProviderType {
    case .native:
        self.defaultWebSocketProviderType = URLSessionWebSocketProvider.self
    case .starscream:
        self.defaultWebSocketProviderType = StarscreamWebSocketProvider.self
    }
}
```
doesn't work since `URLSessionWebSocketProvider.self` is only available in iOS13+, we need the available check

Reply:
🤦‍♂️ I didn't realize this. Sorry. All good then.